src/engine/engine_common.c

```diff
@@ -93,18 +93,43 @@ static inline int _ocf_engine_check_map_entry(struct ocf_cache *cache,
 		return -1;
 }
 
+/* Returns true if core lines no 'entry' and 'entry + 1' within the request
+ * are physically contiguous.
+ */
+static inline bool ocf_engine_clines_phys_cont(struct ocf_request *req,
+		uint32_t entry)
+{
+	struct ocf_map_info *entry1, *entry2;
+	ocf_cache_line_t phys1, phys2;
+
+	entry1 = &req->map[entry];
+	entry2 = &req->map[entry + 1];
+
+	if (!entry1->status || entry1->status == LOOKUP_MISS ||
+			!entry2->status || entry2->status == LOOKUP_MISS)
+	{
```

Comment:
All status codes are non-zero. I think in its current call site the code would sooner BUG-out than `!entry1->status` is `true`

Reply:
also, `entry1->status` can't be true here, so entire check pertaining to entry1 is redundand

Reply:
entry->status may be zero in case of entries which weren't traversed yet

Reply:
The idea of this change is to have correct seq_no value after request is traversed (sets all the statuses to HIT or MISS) and then assigned cachelines in eviction (REMAPPED). During the initial traversation entry2->status will be null.

Reply:
removing all these checks